pygpu/collectives.pyx

```diff
@@ -0,0 +1,299 @@
+from libc.stdlib cimport malloc, calloc, free
+
+from pygpu.gpuarray cimport (gpucontext, GpuContext, _GpuArray, GpuArray,
+                             ensure_context,
+                             GA_NO_ERROR, get_exc, gpucontext_error,
+                             GpuArray_IS_C_CONTIGUOUS,
+                             GA_C_ORDER, GA_F_ORDER, GA_ANY_ORDER,
+                             pygpu_empty_like, pygpu_empty, memcpy)
+from pygpu.gpuarray import GpuArrayException
+
+
+cdef class GpuCommCliqueId:
+    """
+    """
+    def __cinit__(self, GpuContext context=None):
+        self.context = ensure_context(context)
+
+    property comm_id:
```

Comment:
Nccl generates each time a unique id using process id. Will it be useful to pickle a GpuCommCliqueId object in general?